lib/rules/xml-lang-mismatch.json

```diff
@@ -0,0 +1,18 @@
+{
+	"id": "xml-lang-mismatch",
+	"matches": "xml-lang-mismatch-matches.js",
+	"tags": [
+		"cat.language",
+		"wcag2a",
+		"wcag311"
+	],
+	"metadata": {
+		"description": "Ensures that for the html element, there is no mismatch between the primary language in non-empty lang and xml:lang attributes, if both are used",
+		"help": "The value of the primary language subtag for the lang and xml:lang attributes should be the same for the html element"
```

Comment:
I don't expect most users will know what the "primar language subtag" means. I'd suggest we say:

```json
{
  "description": "Ensure that HTML elements with both valid lang and xml:lang attributes agree on the base language of the page",
  "help": "HTML elements with lang and xml:lang must have the same base language"
}
```